group/asg/main.tf

```diff
@@ -14,7 +14,7 @@ resource "aws_autoscaling_group" "asg" {
   max_size                  = "${var.max_size}"
   metrics_granularity       = "${var.metrics_granularity}"
   min_size                  = "${var.min_size}"
-  name                      = "${var.stack_item_label}"
+  name                      = "${var.asg_name_prefix}${var.stack_item_label}"
```

Comment:
I think the override should give you the ability to totally control the value. I would do something like `${length(var.asg_name) > 0 ? var.asg_name : var.stack_item_label}`